packages/rapidpro/src/Adaptor.js

```diff
@@ -49,30 +46,32 @@ export function execute(...operations) {
  * @returns {Operation}
  */
 export function addContact(params, callback) {
-  return state => {
+  return async state => {
     const [resolvedParams] = expandReferences(state, params);
 
     const { host, apiVersion, token } = state.configuration;
 
     const url = `${host}/api/${apiVersion || 'v2'}/contacts.json`;
 
-    const config = {
-      url,
-      data: resolvedParams,
-      headers: { Authorization: `Token ${token}` },
-    };
+    return await fetch(url, {
+      method: 'POST',
+      headers: {
+        Authorization: `Token ${token}`,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(resolvedParams),
+    }).then(async response => {
+      const result = await response.json();
 
-    return http
-      .post(config)(state)
-      .then(response => {
-        console.log('Contact added with uuid:', response.data.uuid);
-        const nextState = {
-          ...composeNextState(state, response.data),
-          response,
-        };
-        if (callback) return callback(nextState);
-        return nextState;
-      });
+      console.log('Contact added with uuid:', result.uuid);
+
+      const nextState = {
```

Comment:
It's hardly worth the effort but I'm going to say it just about is...

If you default callback to `s => s`  up in the parameter declaration, you can just do:
```
return callback({
  ...composeNextState(state, result),
  response,
})
```

Which removes quite a lot of junk code